### ISSUE TITLE
Spindle functionality FOR A FUTURE RELEASE

### ISSUE THREAD

**Issue by Dave288** [grbl/grbl#98]
Some means of spindle speed control (be it use of the third axis to control a virtual or real stepper running a virtual or real pot or PWM or...) would be nice for VFD and DC spindles (getting common on things like 7X lathes and mini mills.)  So spindle on/off, spindle direction, and spindle speed would be desirable.  

Spindle speed and sync might be nice to have on a milling machine, especially for rigid tapping.  The biggest issue is how does one detect the spindle has stabilized at a given speed before starting an operation.

This then gets silly (well, at least difficult) when helical interpolation raises its ugly head for things like thread milling.
